third-party/rsutils/include/rsutils/os/executable-name.h

```diff
@@ -12,8 +12,15 @@ namespace os {
 
 // Returns the path to the executable currently running
 std::string executable_path();
+
+// Returns the filename component of the given path, with or without the extension
+std::string base_name( std::string path, bool with_extension = false );
```

Comment:
`base` is not so clear. You mean `file_name`?

Reply:
In Python it's called the "base" name of the path ([here](https://docs.python.org/3/library/os.path.html#:~:text=os.path.basename(path)%C2%B6)).